### ISSUE TITLE
[1.12-rc3] Allow forcing all swarm tasks to reschedule for a service

### ISSUE THREAD

**Issue by stevvooe** [docker/docker#24469]
On the update of a service, if no changes are made to the task template or the endpoint, no update will be carried out. The service is considered converged at this point no action will be taken by the orchestrator.

There are several cases where one may want to force a re-scheduling of all tasks for a service. A few examples are the following:

1. A user has a change in their cluster that is unobservable to the orchestrator and would like to reschedule all tasks.
2. One or more new nodes has been added and the user wants to rebalance the tasks among them in a controlled fashion.

This issue was partially described in https://github.com/docker/docker/issues/24066#issuecomment-231449232.

The proposal is to add a flag to `docker service update` that would force the service to do a rolling update, without changes:

```
docker service update --reschedule <myservice>
```

Another option would be to make this a top-level command:

```
docker service reschedule <myservice>
```

This approach is similar to how `docker service scale` is implemented.

The implementation would perturb a meaningless value on one the target task spec such that the orchestrator detects a change starts a rolling update.

Tasks required to close this out:

- [ ] Add a field on TaskSpec or ServiceSpec that is perturbed when an update is required (swarmkit).
- [ ] Add field to engine-api
- [ ] Add flag or top-level command to carry out this rescheduling

cc @aluzzardi @aaronlehmann @tiborvass

**Comment by thaJeztah** [comment]
> The implementation would perturb a meaningless value on one the target task spec such that the orchestrator detects a change starts a rolling update.

We could make it a timestamp ("last update") - potentially useful if multiple updates were executed (although I don't know how such situations are handled internally)

**Comment by aaronlehmann** [comment]
Yes, if it was a field like "last forced update", a timestamp could work. Though I suppose a counter would work just as well.

We protect against concurrent updates to the service spec using a version field, so this isn't a situation we have to worry about.

**Comment by stevvooe** [comment]
The key here is have a field that has no meaning. A "last update" field ends up becoming problematic, as it ends up with an ambiguous meaning.
